core/templates/dev/head/components/ExplorationCreationButtonService.js

```diff
@@ -140,31 +140,37 @@ oppia.factory('ExplorationCreationButtonService', [
       showCreateExplorationModal: function(categoryList) {
         alertsService.clearWarnings();
 
-        siteAnalyticsService.registerOpenExplorationCreationModalEvent();
+        var currentPage = window.location.pathname;
 
-        getModalInstance(categoryList, false).result.then(function(result) {
-          var category = $filter('normalizeWhitespace')(result.category);
-          if (!validatorsService.isValidEntityName(category, true)) {
-            return;
-          }
+        if (currentPage.indexOf('/my_explorations') === -1) {
```

Comment:
I think it will be better to do a strict comparison here (i.e. `if (currentPage !== '/my_explorations')`), otherwise this might do the wrong thing if we have a page that just begins with 'my_explorations', like 'my_explorations_stats' or something.